dev/release/docs.md

```diff
@@ -0,0 +1,101 @@
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# How to publish docs
+
+## Overview
+
+We publish our official docs to the following locations:
+
+* https://arrow.apache.org/js/current/ : For the current release
+* https://arrow.apache.org/js/${VERSION} : For the specified version such as `21.0.0`
+* https://arrow.apache.org/js/main/ : For the `main` branch
+
+We also publish docs for fork repositories for preview. It uses the
+following locations:
+
+* https://${GITHUB_ID}.github.io/ : List all available docs
+* https://${GITHUB_ID}.github.io/${BRANCH}/ : For the `${BRANCH}` branch
+
+All of them are automated. This documentation describes how it works.
+
+### apache/arrow-js
+
+apache/arrow-js uses the ASF provided Apache to host generated
+documentation. In general, the `asf-site` branch is used for it. See
+the `publish.whoami` configuration in the top-level `.asf.yaml`.
+
+Note that we don't need to touch the `asf-site` branch manually. It's
+completely maintained automatically. If the `asf-site` branch doesn't
+exist, it's created automatically.
+
+If we update the `asf-site` branch, the `asf-site` branch contents
+will be published to https://arrow.apache.org/js/ automatically.
+
+https://arrow.apache.org/js/ is always redirected to
+https://arrow.apache.org/js/current/ that provides the current release
+documentation. It's implemented by `.htaccess`. `.htaccess` is
+generated by `ci/scripts/update_docs.sh`.
+
+The `asf-site` branch is updated when:
+
+* We push a release tag
+* We push a commit to the `main` branch (We merge a PR to the `main` branch)
+
+Note that the `asf-site` branch isn't updated when we push an RC
+tag. We just build documentation and upload to a GitHub Release when
+we push an RC tag. We add the built documentation to the `asf-site`
+branch when we push a release tag. In this case, `current/` and
+`${VERSION}/` directories in the `asf-site` branch are updated.
+
+We build documentation and add the built documentation to the
+`asf-site` when we push a commit to the `main` branch. In this case,
+`main/` directory in the `asf-site` branch is updated.
+
+See also `ci/scripts/update_docs.sh`.
+
+### Fork repositories
+
+Fork repositories use GitHub Pages to host generated
+documentation. The `gh-pages` branch is used for it.
+
+Note that we don't need to touch the `gh-pages` branch manually. It's
+completely maintained automatically. If the `gh-pages` branch doesn't
+exist, it's created automatically. If your fork is bigger by the
```

Comment:
what does bigger mean here? I don't understand what this means

Reply:
How about "If your fork repository's size becomes bigger by the `gh-pages` branch, you can delete the `gh-pages` branch"?

Reply:
that sounds much better to me! Thanks I wasn't sure what you meant by bigger! Thanks!

Reply:
Applied!